tools/query_verification/src/main/java/com/google/bigquery/Main.java

```diff
@@ -82,9 +89,22 @@ public static void main(String[] args) {
 
         // Data input handling
         if (command.hasOption("d")) {
-            // TODO Data input for data aware verification
+            String[] dataOptionValues = command.getOptionValues("d");
+
+            for (String dataFilePath : dataOptionValues) {
+                String dataFileName = new File(dataFilePath).getName();
+                String dataContents = getContentsOfFile(dataFilePath);
+
+                // Check file format
+                if (dataFileName.substring(dataFileName.lastIndexOf(".") + 1).toLowerCase().equals("csv")) {
```

Comment:
Do we really want to check the extension. I feel sometimes although the files are csv format, they don't necessary have .csv extension. I think maybe we can just assume people are providing valid files?